libnymea-core/hardware/network/networkdevicemonitorimpl.h

```diff
@@ -0,0 +1,73 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+*
+* Copyright 2013 - 2022, nymea GmbH
+* Contact: contact@nymea.io
+*
+* This file is part of nymea.
+* This project including source code and documentation is protected by
+* copyright law, and remains the property of nymea GmbH. All rights, including
+* reproduction, publication, editing and translation, are reserved. The use of
+* this project is subject to the terms of a license agreement to be concluded
+* with nymea GmbH in accordance with the terms of use of nymea GmbH, available
+* under https://nymea.io/license
+*
+* GNU Lesser General Public License Usage
+* Alternatively, this project may be redistributed and/or modified under the
+* terms of the GNU Lesser General Public License as published by the Free
+* Software Foundation; version 3. This project is distributed in the hope that
+* it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+* warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this project. If not, see <https://www.gnu.org/licenses/>.
+*
+* For any further details and any questions please contact us under
+* contact@nymea.io or see our FAQ/Licensing Information on
+* https://nymea.io/license/faq
+*
+* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#ifndef NETWORKDEVICEMONITORIMPL_H
+#define NETWORKDEVICEMONITORIMPL_H
+
+#include <QObject>
+#include <QDateTime>
+
+#include "network/networkdevicemonitor.h"
+
+namespace nymeaserver {
+
+class NetworkDeviceMonitorImpl : public NetworkDeviceMonitor
+{
+    Q_OBJECT
+
+    friend class NetworkDeviceDiscoveryImpl;
```

Comment:
I really dislike all that friending lately. This is quite bad practice. For one, you restrict a implementations name here. We can't have two different implementations, for example to support different platforms.
And more importantly, it encourages spghetti code... there is rarely a good reason to access a private member from outside the class. This creates code where someone working on this class, always needs to fully understand the other class too as it may change private variables at any time without restrictions.

Reply:
Resolved but: The monitor is actually 100% handled by the discovery, so this is a perfect usecase for having a friend class. If you create an different discovery, you need to create a different monitor, in the lib and the resource you would always work with the base class, completely exchangeable.